modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/IgnitePdsDestroyCacheTest.java

```diff
@@ -117,6 +128,135 @@ public void testDestroyCacheOperationNotBlockingCheckpointTest_LocalCache() thro
         doTestDestroyCacheOperationNotBlockingCheckpointTest(true);
     }
 
+    /**
+     * Tests cache destry with hudge dirty pages.
+     */
+    @Test
+    public void testDestroyCache() throws Exception {
+        IgniteEx ignite = startGrid(0);
+
+        ignite.cluster().active(true);
+
+        ignite.getOrCreateCache(new CacheConfiguration<>(DEFAULT_CACHE_NAME).setGroupName("gr1"));
+        ignite.getOrCreateCache(new CacheConfiguration<>("cache2").setGroupName("gr1"));
+
+        try (IgniteDataStreamer<Object, Object> streamer2 = ignite.dataStreamer(DEFAULT_CACHE_NAME)) {
+            PageMemoryEx pageMemory = (PageMemoryEx)ignite.cachex(DEFAULT_CACHE_NAME).context().dataRegion().pageMemory();
+
+            long totalPages = pageMemory.totalPages();
+
+            for (int i = 0; i <= totalPages; i++)
+                streamer2.addData(i, new byte[pageMemory.pageSize() / 2]);
+        }
+
+        ignite.destroyCache(DEFAULT_CACHE_NAME);
+    }
+
+    /**
+     * Tests partitioned cache destry with hudge dirty pages.
+     */
+    @Test
+    public void testDestroyCacheNotThrowsOOMPartitioned() throws Exception {
+        doTestDestroyCacheNotThrowsOOM(false);
+    }
+
+    /**
+     * Tests local cache destry with hudge dirty pages.
+     */
+    @Test
+    public void testDestroyCacheNotThrowsOOMLocal() throws Exception {
+        doTestDestroyCacheNotThrowsOOM(true);
+    }
+
+    /** */
+    public void doTestDestroyCacheNotThrowsOOM(boolean loc) throws Exception {
```

Comment:
I noticed this test does not fail without the fix.

Reply:
i have fail rate 100% without fix

Reply:
After you rework the test, it started working for me too.